packages/moonstone/VirtualList/VirtualListBase.js

```diff
@@ -444,44 +449,18 @@ const VirtualListBaseFactory = (type) => {
 
 			/* For grid lists, find the nearest item from the current item */
 			if (candidateIndex !== -1) {
-				return this.findNearestSpottableItemInExtent(currentIndex, this.getExtentIndex(currentIndex));
+				return noDisabledItem ? candidateIndex : this.findNearestSpottableItemInExtent(candidateIndex, this.getExtentIndex(candidateIndex));
```

Comment:
AFAIK, the first argument of `findNearestSpottableItemInExtent` should be `currentIndex` since that index is used to find the target column in a vertical list for example. Could you check it again please?